### PR TITLE
(fix) - invalidate null link

### PR DESCRIPTION
**Pull request by JoviDeCroock**
## Summary

There was an issue in invalidation when a link was null, this lead to the cache invalidating a record instead of the link since we were checking for a truthy value rather than just undefined.

Fixes: https://github.com/FormidableLabs/urql/issues/693

## Set of changes

- Change link check to `!== undefined` so we account for `null`

### CODE REVIEW

File: ole.log(entity); [hunk 1]
```diff
@@ -0,0 +1,62 @@
+warning: LF will be replaced by CRLF in exchanges/graphcache/src/operations/invalidate.ts.
```
Comment (kitten) [inline ole.log(entity);:1]: Accidental commit?

File: exchanges/graphcache/src/operations/invalidate.ts [hunk 2]
```diff
@@ -18,7 +18,7 @@ export const invalidateEntity = (
 
   for (let i = 0, l = fields.length; i < l; i++) {
     const { fieldKey } = fields[i];
-    if (InMemoryData.readLink(entityKey, fieldKey)) {
+    if (InMemoryData.readLink(entityKey, fieldKey) !== undefined) {
```
Comment (kitten) [inline exchanges/graphcache/src/operations/invalidate.ts:21]: Oops 😅

File: ole.log(entity); (hunk 1 above)
Comment (JoviDeCroock) [reply ole.log(entity);]: Hmm, `eslint` and windows seemed to disagree there did not notice that my bad

**Comment by olistic** [comment]
Thanks for the expedition here! Is there any chance of releasing a patch with this?

**Comment by kitten** [comment]
@olistic I’ll get to releasing this tomorrow 👍 in the meantime you should be able to test this using CodeSandbox CI which publishes temporary packages per PR

**Comment by olistic** [comment]
Thank you @kitten, this is a great tip! 🚀

**Comment by kitten** [comment]
@olistic this has now been released 🌟

**Pull request merged**